materialize-motherduck/config.go

```diff
@@ -173,13 +175,27 @@ func (c *config) db(ctx context.Context) (*stdsql.DB, error) {
 	var tempSecretName string
 	switch c.StagingBucket.StagingBucketType {
 	case stagingBucketTypeS3:
-		createTempSecret = fmt.Sprintf(`CREATE SECRET IF NOT EXISTS (
-			TYPE S3,
-			KEY_ID '%s',
-			SECRET '%s',
-			REGION '%s',
-			SCOPE 's3://%s'
-		);`, c.StagingBucket.AWSAccessKeyID, c.StagingBucket.AWSSecretAccessKey, c.StagingBucket.Region, c.StagingBucket.BucketS3)
+		if c.StagingBucket.Endpoint != "" {
+			// Custom endpoint for S3-compatible storage (e.g. Cloudflare R2).
+			// Use path-style URLs since most S3-compatible services require this.
+			createTempSecret = fmt.Sprintf(`CREATE SECRET IF NOT EXISTS (
+				TYPE S3,
+				KEY_ID '%s',
+				SECRET '%s',
+				REGION '%s',
+				ENDPOINT '%s',
+				URL_STYLE 'path',
+				SCOPE 's3://%s'
+			);`, c.StagingBucket.AWSAccessKeyID, c.StagingBucket.AWSSecretAccessKey, c.StagingBucket.Region, c.StagingBucket.Endpoint, c.StagingBucket.BucketS3)
```

Comment:
It appears the value for `ENDPOINT` should be just the host portion of the URL, without the leading `https://`.
```suggestion
            endpointURL, err := url.Parse(c.StagingBucket.Endpoint)
            if err != nil {
                return nil, err
            }
			// Custom endpoint for S3-compatible storage (e.g. Cloudflare R2).
			// Use path-style URLs since most S3-compatible services require this.
			createTempSecret = fmt.Sprintf(`CREATE SECRET IF NOT EXISTS (
				TYPE S3,
				KEY_ID '%s',
				SECRET '%s',
				REGION '%s',
				ENDPOINT '%s',
				URL_STYLE 'path',
				SCOPE 's3://%s'
			);`, c.StagingBucket.AWSAccessKeyID, c.StagingBucket.AWSSecretAccessKey, c.StagingBucket.Region, endpointURL.Host, c.StagingBucket.BucketS3)
```